contracts/modules/PermissionManager/AutomatedBlackList.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity ^0.4.24;
+import "./GeneralPermissionManager.sol";
+import "../TransferManager/ITransferManager.sol";
+
+/**
+ * @title AutomateBlacklist
+ * @dev to automate blacklist process
+ */
+contract AutomateBlacklist is GeneralPermissionManager,ITransferManager {
```

Comment:
You don't need to inherit the `GeneralPermissionManager` contract. It will automatically available when an issuer will add this contract as a module using the `addModule()` in securityToken. To avail the functionality of permission issuer needs to add the GeneralPermissionManager module after or before adding the `AutomateBlacklistTransferManager` then only you can use the `changePermission()` function to add the perm to delegate corresponds to any module that is present in the module list of the securityToken.

Reply:
An issuer can add `GeneralPermissionManager` after adding the `AutomatedBlacklistTransferManager` but issuer can't use the permission layer until the GPM is added as a module.